docs/template_plugin/tests/functional/op_reference/rnn_sequence.cpp

```diff
@@ -0,0 +1,1036 @@
+            Shape YShape = Shape{batchSize, numDirections, seqLength, hiddenSize};
+            Shape HoShape = Shape{batchSize, numDirections, hiddenSize};
+
+            X = reference_tests::Tensor(XShape, iType, XValues);
+            H_t = reference_tests::Tensor(H_tShape, iType, H_tValues);
+            S_t = reference_tests::Tensor(S_tShape, element::Type_t::i64, S_tValues);
+            W = reference_tests::Tensor(WShape, iType, WValues);
+            R = reference_tests::Tensor(RShape, iType, RValues);
+            B = reference_tests::Tensor(BShape, iType, BValues);
+            Y = reference_tests::Tensor(YShape, oType, YValues);
+            Ho = reference_tests::Tensor(HoShape, oType, HoValues);
+        }
+
+    size_t batchSize;
+    size_t inputSize;
+    size_t hiddenSize;
+    size_t seqLength;
+    size_t numDirections;
+    float clip;
+    op::RecurrentSequenceDirection rnn_direction;
+    element::Type_t iType;
+    element::Type_t oType;
+
+    reference_tests::Tensor X;
+    reference_tests::Tensor H_t;
+    reference_tests::Tensor S_t;
+    reference_tests::Tensor W;
+    reference_tests::Tensor R;
+    reference_tests::Tensor B;
+    reference_tests::Tensor Y;
+    reference_tests::Tensor Ho;
+    std::string testcaseName;
+};
+
+class ReferenceRNNSequenceTest : public testing::TestWithParam<RNNSequenceParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.X.data, params.H_t.data, params.S_t.data, params.W.data, params.R.data, params.B.data};
+        refOutData = {params.Y.data, params.Ho.data};
+    }
+
+    static std::string getTestCaseName(const testing::TestParamInfo<RNNSequenceParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "iType=" << param.iType << "_";
+        result << "xShape=" << param.X.shape << "_";
+        result << "htShape=" << param.H_t.shape << "_";
+        result << "stShape=" << param.S_t.shape << "_";
+        result << "wShape=" << param.W.shape << "_";
+        result << "rShape=" << param.R.shape << "_";
+        result << "bShape=" << param.B.shape << "_";
+        result << "YShape=" << param.Y.shape << "_";
+        result << "hoShape=" << param.Ho.shape << "_";
+        result << "clip=" << param.clip << "_";
+        result << "RNNdirection=" << param.rnn_direction;
+        if (!param.testcaseName.empty())
+            result << "_" << param.testcaseName;
+
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Model> CreateFunction(const RNNSequenceParams& params) {
+        const auto X = std::make_shared<op::v0::Parameter>(params.X.type, params.X.shape);
+        const auto H_t = std::make_shared<op::v0::Parameter>(params.H_t.type, params.H_t.shape);
+        const auto S_t = std::make_shared<op::v0::Parameter>(params.S_t.type, params.S_t.shape);
```

Comment:
usually only states are suffixed with _t, because in the same way they are written in the equations
I don't remember anyone using S_t to specify sequence lenghts in articles, may be it is better to rename S_t to seq_len or sequence_lenghts or s_lenghts, something like that
the same for GRUSequence

Reply:
Renamed them! Thank you for the review!